.github/workflows/zfs-qemu.yml

```
@@ -6,15 +6,48 @@ on:
 
 jobs:
 
+  test-config:
+    name: Generate configuration for QEMU test
+    runs-on: ubuntu-24.04
+    outputs:
+      test_os: ${{ steps.os.outputs.os }}
+      ci_type: ${{ steps.os.outputs.ci_type }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Generate OS config and CI type
+        id: os
+        run: |
+          FULL_OS='["almalinux8", "almalinux9", "centos-stream9", "debian11", "debian12", "fedora39", "fedora40", "freebsd13", "freebsd13r", "freebsd14", "freebsd14r", "ubuntu20", "ubuntu22", "ubuntu24"]'
+          QUICK_OS='["almalinux8", "almalinux9", "debian12", "freebsd13", "freebsd14", "ubuntu22"]'
```

```suggestion
          QUICK_OS='["almalinux8", "almalinux9", "debian12", "fedora40", "freebsd13", "freebsd14", "ubuntu24"]'
```

Let's  include the latest Fedora which is most likely to have a current kernel and the latest Ubuntu LTS instead of the previous one.